pkg/adapter/mtping/runner.go

```diff
@@ -101,14 +101,33 @@ func (a *cronJobsRunner) AddSchedule(source *sourcesv1.PingSource) cron.EntryID
 	ctx = observability.WithSpanData(ctx, spanName, int(trace.SpanKindProducer),
 		observability.K8sAttributes(source.Name, source.Namespace, sourcesv1.Resource("pingsource").String()))
 
-	schedule := source.Spec.Schedule
-	if source.Spec.Timezone != "" {
-		schedule = "CRON_TZ=" + source.Spec.Timezone + " " + schedule
-	}
-
 	ctx = kncloudevents.ContextWithMetricTag(ctx, metricTag)
-	id, _ := a.cron.AddFunc(schedule, a.cronTick(ctx, event))
-	return id
+	// If Date is not set, we use Schedule to trigger cron job.
+	if source.Spec.Date == "" {
+		schedule := source.Spec.Schedule
+		if source.Spec.Timezone != "" {
+			schedule = "CRON_TZ=" + source.Spec.Timezone + " " + schedule
+		}
+		id, _ := a.cron.AddFunc(schedule, a.cronTick(ctx, event))
+		return id
+
+	} else {
+		// Date is specified to fire events only once.
+		dateString := source.Spec.Date
+		date, err := time.Parse("2006-01-02 15:04:05", dateString)
```

Comment:
why this non-standard format? I would use this: https://262.ecma-international.org/5.1/#sec-15.9.1.15

Reply:
ok, I will use this format